results2junit

```diff
@@ -0,0 +1,289 @@
+#!/usr/bin/env python
+#
+# results2junit - convert docker-autotest results file to Jenkins format
+#
+"""
+Convert docker-autotest results file (named 'status') to a results.junit
+file suitable by Jenkins.
+"""
+
+import argparse
+import os
+import re
+import sys
+import time
+
+
+def xml_escape(input_string):
+    """
+    Escape XML-sensitive characters in a string
+    """
+    if not input_string:
+        return ''
+    if isinstance(input_string, basestring):
+        s_out = input_string
+    else:
+        s_out = '%s' % input_string
+
+    s_out = s_out.replace('&', '&amp;')
+    s_out = s_out.replace('<', '&lt;')
+    s_out = s_out.replace('>', '&gt;')
+    s_out = s_out.replace('"', '&quot;')
+    return s_out
+
+
+class TestSuite(object):
+    """
+    Set of test results, initialized from a list read by blah blah
+    """
+
+    def __init__(self, name, results):
+        self.input_name = name
+        # Strip gobbledygook from rhel-latest-7-3-ADEPT-full-26-a451b130bf40
+        self.name = re.sub(r'-\d+-[0-9a-f]{11,}$', '', name)
+        self.results = results
+        self.count = {'tests': 0, 'failures': 0, 'errors': 0, 'skipped': 0}
```

Comment:
I guess I don't understand this comment. I will try to explain the code, and if that doesn't address your concern, please let me know.

The `status` file contains the following status strings: GOOD, TEST_NA, ERROR, and FAIL. `results.junit`, however, needs to have different strings: Jenkins expects `skipped` (for TEST_NA), `errors` (plural) for ERROR, and `failures` (not just fail) for FAIL. I need to convert the `status` strings to the form expected by Jenkins. Does that address your question?

I've refactored that code to reduce some duplication and, I hope, make it easier to understand. Coming in next commit.